### PR TITLE
Document Service#afterCommit handler [ECR-2728]

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
### Overview
Adds documentation about `Service#afterCommit` usage. 

- [x] Rebase onto master once #248 is integrated
- [x] ~~Add proper Javadoc links once they are published~~ — delayed till we publish them

### CODE REVIEW

**Comment by dmitry-timofeev** [comment]
@slowli , Shall we create a separate branch for pre-EJB-release documentation, because core and EJB releases are not synchronized?

File: src/get-started/java-binding.md [hunk 1]
```diff
@@ -180,11 +197,44 @@ current storage state when the transaction is executed. Exonum passes `Fork`
 as an argument – a view that allows performing modifying operations. A service
 schema object can be used to access data collections of this service.
 
+Also, `Transaction#execute` method may throw `TransactionExecutionException`
+which contains a transaction error report. This feature allows users to notify
+Exonum about an error in a transaction execution whenever one occurs.
+It may check the preconditions before executing a transaction and either
```
Comment (Maria-Nosyk) [inline src/get-started/java-binding.md]: It checks the preconditions...

File: src/get-started/java-binding.md [hunk 2]
```diff
@@ -180,11 +197,44 @@ current storage state when the transaction is executed. Exonum passes `Fork`
...
+accepts it or throws an exception that is further transformed into an Exonum
+core [TransactionResult enum][transaction-result] containing an error code and
+a message with error data.
+If transaction execution fails, the changes made by the transaction are
+rolled back, while the error data is stored in the database for further user
+reference. Light clients also provide access to information on the
+[transaction][exonum-transaction] execution result
+(which may be either success or failure) to their users.
+
+### Blockchain events
```
Comment (Maria-Nosyk) [inline src/get-started/java-binding.md]: Events

File: src/get-started/java-binding.md [hunk 3]
```diff
@@ -180,11 +197,44 @@ current storage state when the transaction is executed. Exonum passes `Fork`
...
+
+A service can also handle a block commit event, that occurs each time
```
Comment (Maria-Nosyk) [inline src/get-started/java-binding.md]: delete comma

File: src/get-started/java-binding.md [hunk 4]
```diff
@@ -362,6 +412,25 @@ service:
 - Follow the instructions in the [application guide][app-tutorial] to configure
   and start an Exonum node with your service.
 
+## Common Library
+
+Java Binding includes a library module that can be useful for Java client
+applications that interact with an Exonum service and
+does not have the dependency on Java Binding Core. The module contains Java
```
Comment (Maria-Nosyk) [inline src/get-started/java-binding.md]: Java Binding includes a library module that can be useful for Java client
applications that interact with an Exonum service. The module does not have the dependency on Java Binding Core, but it contains Java classes obligatory for core that can now as well be easily applied in clients, if necessary.

File: src/get-started/java-binding.md (hunk 1 above)
Comment (dmitry-timofeev) [reply src/get-started/java-binding.md]: I'd keep it as is (or clarify), as a _transaction_ may (or may not) check the preconditions.

File: src/get-started/java-binding.md [hunk 5]
```diff
@@ -222,6 +222,26 @@ reference. Light clients also provide access to information on the
...
+implementations of [`Service#afterCommit(BlockCommittedEvent)`][service-after-commit]
+in each deployed service. Each node in the network processes that event
+independently from other nodes. The event includes a `Snapshot`,
+allowing a read-only access to the database state _exactly_ after the commit
+of the corresponding block.
+
+As services can read the database state in the handler, they may detect
+any changes in it, e.g., that a certain transaction is executed;
+or some condition is met. Services may also create and submit new transactions
+using [`Node#submitTransaction`][node-submittransaction]. Using this callback
```
Comment (MakarovS) [inline src/get-started/java-binding.md]: Nit: why do we use `submittransaction` here, but `after-commit` above? (I like hyphens better)

File: src/get-started/java-binding.md [hunk 6]
```diff
@@ -222,6 +222,26 @@ reference. Light clients also provide access to information on the
 [transaction][exonum-transaction] execution result
 (which may be either success or failure) to their users.
 
+### Blockchain Events
+
+A service can also handle a block commit event that occurs each time
```
Comment (bullet-tooth) [inline src/get-started/java-binding.md]:
```suggestion
A service may handle a block commit event that occurs each time
```

File: src/get-started/java-binding.md [hunk 7]
```diff
@@ -222,6 +222,26 @@ reference. Light clients also provide access to information on the
...
+the framework commits a new block. The framework delivers this event to
+implementations of [`Service#afterCommit(BlockCommittedEvent)`][service-after-commit]
```
Comment (bullet-tooth) [inline src/get-started/java-binding.md]: such as "callback" word is used below:
"implementations of the callback method..."

File: src/get-started/java-binding.md [hunk 8]
```diff
@@ -407,9 +427,9 @@ service:
 ## Common Library
 
 Java Binding includes a library module that can be useful for Java client
-applications that interact with an Exonum service and
-does not have the dependency on Java Binding Core. The module contains Java
-classes obligatory for core that can now as well be easily applied in clients,
+applications that interact with an Exonum service. The module does not
+have the dependency on Java Binding Core, but it contains Java classes
+obligatory for core that can now as well be easily used in clients,
```
Comment (bullet-tooth) [inline src/get-started/java-binding.md]:
```suggestion
obligatory for the core that can now as well be easily used in clients,
```

File: src/get-started/java-binding.md (hunk 6 above)
Comment (dmitry-timofeev) [reply src/get-started/java-binding.md]: I'd keep can, as it seems to fit here better (describing a capability)

**Comment by dmitry-timofeev** [comment]
Does anyone know what is that:
```
./src/get-started/java-binding.md: 484: MD034/no-bare-urls Bare URL used [Context: "https://exonum.com/doc/api/jav..."]
```

We have had such for ages, haven't we?

**Comment by slowli** [comment]
@dmitry-timofeev It's Markdown linter not understanding the following syntax:

```markdown
[foo]: https://foo.com/ <!-- foo -->
```

You should change it to

```markdown
<!-- foo -->
[foo]: https://foo.com/
```

 to be safer.

**Comment by dmitry-timofeev** [comment]
@slowli , Thank you, let’s see if it works!

**Pull request merged**